src/java.base/share/classes/java/util/Properties.java

```diff
@@ -921,6 +942,19 @@ private void store0(BufferedWriter bw, String comments, boolean escUnicode)
         bw.flush();
     }
 
+    private static void writeDateComment(BufferedWriter bw) throws IOException {
+        // value of java.util.Properties.storeDate system property isn't sensitive
+        // and so doesn't need any security manager checks to make the value accessible
+        // to the callers
+        String storeDate = StaticProperty.javaUtilPropertiesStoreDate();
+        if (storeDate != null && !storeDate.isEmpty()) {
+            writeComments(bw, storeDate);
+        } else {
+            bw.write("#" + DateTimeFormatter.ofPattern(dateFormatPattern).format(ZonedDateTime.now()));
+            bw.newLine();
+        }
```

Comment:
Since there is no longer a need to format an arbitrary date, I'd suggest going back to the original Date.toString() code. It removes the need to replicate the format using DateTimeBuilder and is known to be the same as before.

If you keep the DateTimeFormat version, you would want "uuuu" instead of "yyyy". In java.time.
DateTimeFormatterBuilder uses slightly different pattern letter conventions as compared to  SimpleDateFormat.

Reply:
> Since there is no longer a need to format an arbitrary date, I'd suggest going back to the original Date.toString() code. It removes the need to replicate the format using DateTimeBuilder and is known to be the same as before.

Done. I pushed an update to the PR which switches back to using Date.toString() for the date comment. It also does a minor adjustment to the javadoc to clarify this behaviour.